src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -67,106 +65,93 @@ public StudyService(
-        Study study = new Study(studyName, networkInfos.getNetworkUuid(), networkInfos.getNetworkId(), description, caseFormat, caseUUid, true);
-        studyRepository.insert(study);
+    Mono<Study> createStudy(String studyName, MultipartFile caseFile, String description) {
+        Mono<UUID> caseUUid;
+        try {
+            caseUUid = importCase(caseFile);
+        } catch (IOException e) {
+            return Mono.error(new StudyException("error when importing the case"));
+        }
+        return caseUUid.flatMap(uuid -> {
+            Mono<NetworkInfos> networkInfos = persistentStore(uuid);
+            Mono<String> caseFormat = getCaseFormat(uuid);
+            return Mono.zip(networkInfos, caseFormat)
+                    .flatMap(t -> {
+                        Study study = new Study(studyName, t.getT1().getNetworkUuid(), t.getT1().getNetworkId(), description, t.getT2(), uuid, true);
+                        return studyRepository.insert(study);
+                    });
+        });
     }
 
-    Study getStudy(String studyName) {
-        return studyRepository.findByName(studyName).orElse(null);
+    Mono<Study> getStudy(String studyName) {
+        return studyRepository.findByName(studyName);
     }
 
-    void deleteStudy(String studyName) {
-        Optional<Study> studyOpt = studyRepository.findByName(studyName);
-        if (studyOpt.isEmpty()) {
-            return;
-        }
-        Study study = studyOpt.get();
-        if (study.isCasePrivate()) {
-            try {
-                caseServerRest.delete("/" + CASE_API_VERSION + "/cases/{caseUuid}", study.getCaseUuid());
-            } catch (HttpStatusCodeException e) {
-                throw new StudyException("deleteStudy HttpStatusCodeException", e);
-            }
-        }
-        studyRepository.deleteByName(studyName);
-    }
+    Mono<Void> deleteStudy(String studyName) {
+        Mono<Study> studyMono = studyRepository.findByName(studyName);
 
-    List<CaseInfos> getCaseList() {
-        ParameterizedTypeReference<List<CaseInfos>> parameterizedTypeReference = new ParameterizedTypeReference<List<CaseInfos>>() { };
-        ResponseEntity<List<CaseInfos>> responseEntity;
-        try {
-            responseEntity = caseServerRest.exchange("/" + CASE_API_VERSION + "/cases",
-                    HttpMethod.GET,
-                    null,
-                    parameterizedTypeReference);
-
-            if (responseEntity.getStatusCode() == HttpStatus.OK) {
-                return responseEntity.getBody();
-            } else {
-                return Collections.emptyList();
+        return studyMono.flatMap(study -> {
+            String path = UriComponentsBuilder.fromPath(DELIMITER + CASE_API_VERSION + "/cases/{caseUuid}")
+                    .buildAndExpand(study.getCaseUuid())
+                    .toUriString();
+
+            if (study.isCasePrivate()) {
+                return webClient.delete()
```

Comment:
If i understand correctly, when the case is private we don't delete the study ? This is a bug, and there should be a test to show the bug

Reply:
if the case is public we don't delete it *  but we delete the study.